pkg/oauth/handlers/authenticator.go

```diff
@@ -101,24 +102,30 @@ func (h *accessAuthenticator) HandleAccess(ar *osin.AccessRequest, w http.Respon
 		klog.Warningf("Received unknown access token type: %s", ar.Type)
 	}
 
+	audit.AddUsernameAnnotation(ar.HttpRequest, ar.Username)
 	if err != nil {
+		audit.AddDecisionAnnotation(ar.HttpRequest, audit.ErrorDecision)
 		klog.V(4).Infof("Unable to authenticate %s: %v", ar.Type, err)
 		return err
 	}
+	if !ok {
+		audit.AddDecisionAnnotation(ar.HttpRequest, audit.DenyDecision)
 
-	if ok {
-		// Disable refresh_token generation
-		ar.GenerateRefresh = false
-		ar.Authorized = true
-		if info != nil {
-			// TODO something with audiences?
-			ar.AccessData.UserData = info.User
-		}
+		return nil
+	}
+	audit.AddDecisionAnnotation(ar.HttpRequest, audit.AllowDecision)
 
-		if e, ok := ar.Client.(TokenMaxAgeSeconds); ok {
-			if maxAge := e.GetTokenMaxAgeSeconds(); maxAge != nil {
-				ar.Expiration = *maxAge
-			}
+	// Disable refresh_token generation
+	ar.GenerateRefresh = false
+	ar.Authorized = true
+	if info != nil {
+		// TODO something with audiences?
```

Comment:
sigh 😞

Reply:
I didn't do it. I just shuffled it around. :)

Reply:
@s-urbaniak, should we create a story for that?

Reply:
@ibihim i would say so yes, at least to assert the context